app/bundles/LeadBundle/Command/ContactScheduledExportCommand.php

```diff
@@ -55,17 +59,27 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 
         $count = 0;
 
-        foreach ($contactExportSchedulers as $contactExportScheduler) {
-            $contactExportSchedulerEvent = new ContactExportSchedulerEvent($contactExportScheduler);
-            $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::CONTACT_EXPORT_PREPARE_FILE);
-            $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::CONTACT_EXPORT_SEND_EMAIL);
-            $this->eventDispatcher->dispatch($contactExportSchedulerEvent, LeadEvents::POST_CONTACT_EXPORT_SEND_EMAIL);
-            ++$count;
-        }
+        $this->processSignalService->registerSignalHandler(
+            fn (int $signal) => $output->writeln(sprintf('Signal %d caught.', $signal))
+        );
```

Comment:
Let's move registering of the signal handler to the very beginning of the `execute()` method (before possibly slow DB operations). The earlier the handler is registered the better (we minimize situations when the handler is registered to late).